documentation/specs/BuildCheck/CustomBuildCheck.md

```diff
@@ -0,0 +1,144 @@
+# MSBuild Custom Build Checks Guide
+
+## Overview
+Custom build checks in MSBuild provide developers with an extensibility point to implement specific validation and reporting during the build process.
+
+## Step-by-Step Custom Check Creation
+
+### 1. Understand the Motivation
+Before creating a custom check, identify your specific project needs, e.g.:
+- Do you need to enforce version constraints?
+- Are there security concerns with certain configurations?
+- Do you want to maintain consistent project properties?
+Depending on this, different MSBuild project stages can be considered (Evaluation, Build or Post Build events.)
+
+### 2. Install Prerequisites
+- Install .NET SDK 9 or higher
+
+- Install MSBuild Custom Check Template
+```powershell
+dotnet new install Microsoft.Build.Templates
+```
+
+### 3. Create Project Template
+```powershell
+dotnet new msbuildcheck -n MyCustomBuildChecks
+```
+
+### 4. Examine Template Structure
+- Inherit from the MSBuild API base class
+- Do NOT attempt to unbind from the base class Check because it's validated on MSBuild side
+- Follow the strict type checking requirements
```

Comment:
Are all those 3 refering to the same thing?

I'd just put something like 'Inherit your checks from the `Microsoft.Build.Experimental.BuildCheck.Check` base class as already done by the template as otherwise your checks won't be registered during build runtime.'